server/service/IndividualServicesService.js

```diff
@@ -11084,6 +11086,54 @@ exports.regardControllerAttributeValueChange = function (url, body, user, origin
   });
 }
 
+// Map of per-mountname locks
+const locks = new Map();
+const queueCounts = new Map(); // track queue depth per mountname
+
+/**
+ * Ensures only one update per mountname runs at a time.
+ * id String mountname
+ * alarmBody V1_regarddevicealarm_body
+ * fn Function async function that performs the update
+ * returns {Promise<any>} result of fn
+ */
+async function withLock(id, alarmBody, fn) {
+  if (!locks.has(id)) {
+    locks.set(id, Promise.resolve());
+    queueCounts.set(id, 0);
+  }
+  const prev = locks.get(id);
+  queueCounts.set(id, (queueCounts.get(id) || 0) + 1);
+  const queueSize = queueCounts.get(id);
+
+  logAlarmNotificationUpdate(`***************************************START - Processing alarm for Mountname=${id}***************************************`);
+  logAlarmNotificationUpdate(`Mountname=${id} queued update. Queue size now: ${queueSize}`);
+  logAlarmNotificationUpdate(`Alarm Mountname=${id} to be updated: ${alarmBody}`);
+  logger.info(`Mountname=${id} queued update. Queue size now: ${queueSize}`);
+
+  let release;
+  const p = new Promise((res) => (release = res));
+  locks.set(id, prev.then(() => p));
+
+  try {
+    const start = Date.now();
+    const result = await fn();
+    const duration = (Date.now() - start)/1000;
+
+    logAlarmNotificationUpdate(
+      `Mountname=${id} processed update. Duration=${duration}s. Queue size after process: ${queueSize - 1}`
+    );
+    logAlarmNotificationUpdate(`Mountname=${id} updated for: ${alarmBody}`);
+    logger.info(`Mountname=${id} processed update. Duration=${duration}s. Queue size after process: ${queueSize - 1}`);
+    logAlarmNotificationUpdate(`***************************************END - Processing alarm for Mountname=${id} completed***************************************`);
+
+    return result;
+  } finally {
+    release();
+    queueCounts.set(id, queueCounts.get(id) - 1);
```

Comment:
We're storing a Promise and a queue count for each mountname (id) in locks and queueCounts, but never removing them after processing is complete. 
As MWDI already have memory consumption problems, suggesting to clean up both maps when the queue count drops to zero. Thanks.
```js
finally {
  release();
  const newCount = queueCounts.get(id) - 1;
  if (newCount <= 0) {
    queueCounts.delete(id);
    locks.delete(id);
  } else {
    queueCounts.set(id, newCount);
  }
}

```